chia/_tests/core/mempool/test_mempool.py

```diff
@@ -2866,7 +2866,7 @@ async def test_invalid_coin_spend_coin(
         new_bundle = recursive_replace(spend_bundle, "coin_spends", [coin_spend_0, *spend_bundle.coin_spends[1:]])
         assert spend_bundle is not None
         res = await full_node_1.full_node.add_transaction(new_bundle, new_bundle.name(), test=True)
-        assert res == (MempoolInclusionStatus.FAILED, Err.INVALID_SPEND_BUNDLE)
+        assert res == (MempoolInclusionStatus.FAILED, Err.WRONG_PUZZLE_HASH)
```

Comment:
this seems suspicious did the test change ?

Reply:
I believe the mempool, currently, validates the puzzle reveal against the puzzle hash really late, here: https://github.com/Chia-Network/chia-blockchain/blob/main/chia/full_node/mempool_manager.py#L635-L637

With the new `chia_rs` we check it early, before running the puzzle. And we have a more precise error code.